internal/cloud/azureshared/appcredentials.go

```diff
@@ -37,8 +40,16 @@ func ApplicationCredentialsFromURI(cloudServiceAccountURI string) (ApplicationCr
 		return ApplicationCredentials{}, fmt.Errorf("invalid service account URI: invalid host: %s", uri.Host)
 	}
 	query := uri.Query()
+
+	subscriptionPattern := regexp.MustCompile(`subscriptions/([^/]+)/`)
```

Comment:
`regexp.MustCompile` always go to the top of the file as `var`.

Reply:
is there a reason?